### PR TITLE
Improve 'new', 'add', 'push', and 'pull' command help texts 

### PR DESCRIPTION
**Pull request by sameerajayasoma**
Added more documentation with a refreshed formatting inspired by Go and Git help pages. 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
 SYNOPSIS
        ballerina add <module-name> [-t <template-name> | --template <template-name>]
-                     [-l | --list]
+       ballerina add [-l | --list]
 
 
 DESCRIPTION
-       Add a new module to a Ballerina project. The command should
-       be run inside a Ballerina project.
+       Add creates a new directory and initializes Ballerina module inside
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help]:
```suggestion
       Add creates a new directory and initializes a Ballerina module inside
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help [hunk 2]
```diff
@@ -1,31 +1,42 @@
...
+       the 'src' directory of the project. The command should be run inside
+       the project root.
 
        A template can also be specified when creating the new module.
+       If a template is not specified, this command creates a module
+       with a 'main' function.
+
+       There are two built-in templates named: 'main' and 'service'.
+
+       You can download templates from Ballerina central as well.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help]:
```suggestion
       You can download templates from Ballerina Central as well.
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help [hunk 3]
```diff
@@ -1,31 +1,42 @@
...
+       In that case <template-name> becomes a qualified module name.
+
+           <org-name>/<module-name>[:<version>]
 
 
 OPTIONS
-       -t, --template
-           Add a module using a predefined template. The
-           default template is a module with a main.
+       -t, --template <template-name>
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-add.help]:
```suggestion
       -t <template-name>, --template <template-name>
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-pull.help [hunk 4]
```diff
@@ -1,25 +1,26 @@
...
+         Ballerina Central is a module repository hosted at
+         https://central.ballerina.io/. A module repository organizes modules
+         into a three-level hierarchy: organization, module name, and version.
+         Organizations are unique within a repository and can be mapped to an
+         individual user or organization registered with the repository.
 
 EXAMPLES
-         Pull the Gmail connector by WSO2 from Ballerina Central
-         $ ballerina pull wso2/gmail
+         Pull the latest version of 'gmail' connector in `wso2` organization
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-pull.help]:
```suggestion
         Pull the latest version of 'gmail' connector in 'wso2' organization
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-pull.help [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
+         from Ballerina Central
+            → ballerina pull wso2/gmail
 
-         Pull Bar module by Foo organization from Ballerina Central
-         $ ballerina pull foo/bar
+         Pull the '1.1.0' version of 'gmail' connector in `wso2` organization
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-pull.help]:
```suggestion
         Pull the '1.1.0' version of 'gmail' connector in 'wso2' organization
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help [hunk 6]
```diff
@@ -1,32 +1,37 @@
...
-       ballerina push <module-name> [--no-build] [--repository home]
+       ballerina push <module-name>
+       ballerina push (-a | --all)
 
 
 DESCRIPTION
-       Push the given module to Ballerina Central
-       (https://central.ballerina.io/) repository.
+       Push the specified module to Ballerina Central repository or
+       if '-a' or '--all' option is specified, then pull all the modules
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help]:
```suggestion
       if '-a' or '--all' option is specified, then push all the modules
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help [hunk 7]
```diff
@@ -1,32 +1,37 @@
...
+       in the current project.
 
-       By default, the sources will be built before pushing
+       By default, the Push command build the sources before pushing
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help]:
```suggestion
       By default, the push command builds the sources before pushing
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help [hunk 8]
```diff
@@ -1,32 +1,37 @@
...
        the module to Ballerina Central.
 
-       Before pushing modules to central,
+       Before pushing modules to central:
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help]:
```suggestion
       Before pushing modules to Ballerina Central:
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help [hunk 9]
```diff
@@ -1,32 +1,37 @@
...
+       1) Create an organization by registering with Ballerina central at
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-push.help]:
```suggestion
       1) Create an organization by registering with Ballerina Central at
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-build.help [hunk 10]
```diff
@@ -39,26 +38,26 @@ OPTIONS
...
-       <project-root>/target/bin directory.
+       Compile a Ballerina source file and create an executable jar.
+          → ballerina build hello.bal
+
+       Build the 'hello' module. This will generate a hello.jar file
+       in <project-root>/target/bin directory.
+          → ballerina build hello
+
+
+       Build all the modules that have entry points in a project in to
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-build.help]:
```suggestion
       Build all the modules that have entry points in a project into
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-build.help [hunk 11]
```diff
@@ -39,26 +38,26 @@ OPTIONS
...
+       executables. This will generate executable jar files and place
+       them in '<project-root>/target/bin' directory.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-build.help]:
```suggestion
       them in the '<project-root>/target/bin' directory.
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help [hunk 12]
```diff
@@ -1,46 +1,48 @@
...
-       module or a ballerina file.
+       Doc builds documentation for a given module or a ballerina file.
 
-       To generate docs for all modules of current project, use -a option.
+       To build documentation for all modules of the current project,
+       use -a or -all option.
 
-       API Documentation will be written to target/apidocs folder in
-       projects.
+       This command writes the dodule documentation to the 'target/apidocs'
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help]:
```suggestion
       This command writes the module documentation to the 'target/apidocs'
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help [hunk 13]
```diff
@@ -1,46 +1,48 @@
...
        If docs are being generated for a single ballerina file,
        output path should be provided with -o option.
 
+
 OPTIONS
        -o <output>
-           Writes API Documentation to the given folder. This is required
-           if you are generating docs for a ballerina file outside of a
-           project.
+           Build and store the module documentation in the spefified directory.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help]:
```suggestion
           Build and store the module documentation in the specified directory.
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help [hunk 14]
```diff
@@ -1,46 +1,48 @@
...
+           This is required if you are generating docs for a ballerina file
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-doc.help]:
```suggestion
           This is required if you are generating docs for a Ballerina file
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-format.help [hunk 15]
```diff
@@ -20,38 +19,21 @@ DESCRIPTION
...
-       Formats a Ballerina file.
-       $ ballerina format hello.bal
+       Format a Ballerina file.
+          → ballerina format helloworld.bal
 
-       Performs a dry run to see which files will be formatted.
-       $ ballerina format -d
\ No newline at end of file
+       Perform a dry run to see which files will be formatted.
+          → ballerina format -d
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-format.help]:
```suggestion
          → ballerina format -d

```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 16]
```diff
@@ -1,34 +1,39 @@
...
 
        --output <path>
-           [ Optional ] Location of the generated Ballerina source files.
-           If output path not specified, output will be written to a directory correspond to the package in protocol buffer definition.
-           If package not specified, output will be written to a `temp` directory in the current location.
-
-       --mode <client-or-server>
-           [ Optional ] Set mode (client or server) to generate sample codes.
+           Location of the generated Ballerina source files.
+           If output path not specified, output will be written to a directory
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
           If output path is not specified, output will be written to a directory
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 17]
```diff
@@ -1,34 +1,39 @@
...
+           correspond to the package in protocol buffer definition.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
           corresponding to the package in the protocol buffer definition.
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 18]
```diff
@@ -1,34 +1,39 @@
...
+           If package not specified, output will be written to a `temp`
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
           If package is not specified, output will be written to a `temp`
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 19]
```diff
@@ -1,34 +1,39 @@
...
+           directory in the current location.
+
+       --mode client | server
+           Set mode (client or server) to generate sample codes.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
           Set mode (client or server) to generate sample code.
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 20]
```diff
@@ -1,34 +1,39 @@
...
            If not specified, only the stub file is generated.
 
 EXAMPLES
-       Generate the Ballerina gRPC stub file for the given .proto file to a `stub` directory:
-       $ ballerina grpc --input chat.proto --output stub
+       Generate the Ballerina gRPC stub file for the given .proto file to a
+       `stub` directory:
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
       'stub' directory:
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 21]
```diff
@@ -1,34 +1,39 @@
...
+          → ballerina grpc --input chat.proto --output stub
 
-       Generate the Ballerina gRPC stub file and client sample code for the given .proto file to a `client` directory:
-       $ ballerina grpc --input chat.proto --output client --mode client
+       Generate the Ballerina gRPC stub file and client sample code for the
+       given .proto file to a `client` directory:
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
       given .proto file to a 'client' directory:
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help [hunk 22]
```diff
@@ -1,34 +1,39 @@
...
+          → ballerina grpc --input chat.proto --output client --mode client
 
-       Generate the Ballerina gRPC stub file and service sample code for the given .proto file to a `service` directory:
-       $ ballerina grpc --input chat.proto --output service --mode service
+       Generate the Ballerina gRPC stub file and service sample code for the
+       given .proto file to a `service` directory:
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-grpc.help]:
```suggestion
       given .proto file to a 'service' directory:
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-search.help [hunk 23]
```diff
@@ -1,27 +1,12 @@
...
-EXAMPLES
-       Search for Foo organization
-       $ ballerina search Foo
-
-       Search for Gmail connector
-       $ ballerina search Gmail
-
-       Search for modules from WSO2
-       $ ballerina search wso2
+       The keyword is treated as case-insensitive when search is executed.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-search.help]:
```suggestion
       The keyword is treated as case-insensitive when search is executed.

```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help [hunk 24]
```diff
@@ -1,16 +1,13 @@
...
-       ballerina test - execute Ballerina tests
+       ballerina-test - Run module tests
 
 SYNOPSIS
-       ballerina test [--offline]
-                      [--sourceroot <path>]
-                      [--experimental] [--skip-lock]
-                      [--config <config_file>]
-                      [<module-name> | -a | --all]
+       ballerina build [<options>] <module-name> [(--key=value)...]
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help]:
```suggestion
       ballerina test [<options>] <module-name> [(--key=value)...]
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help [hunk 25]
```diff
@@ -1,16 +1,13 @@
...
+       ballerina build -a | --all [<options>] [(--key=value)...]
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help]:
```suggestion
       ballerina test -a | --all [<options>]
```

Key value pairs don't seem to be working atm, after -a/--all.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-version.help [hunk 26]
```diff
@@ -1,23 +1,18 @@
...
+       ballerina-version - Print the Ballerina version
 
 SYNOPSIS
-       ballerina <-v | --version | version>
+       ballerina version
 
 
 DESCRIPTION
-       Print the current Ballerina version in use.
+       Version prints the Ballerina distribution version, supported
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-version.help]:
```suggestion
       Version prints the Ballerina distribution version and the 
```

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-version.help [hunk 27]
```diff
@@ -1,23 +1,18 @@
...
+       language specification version, and the Ballerina tool version.
```
Comment (MaryamZi) [inline cli/ballerina-tool/src/main/resources/cli-help/ballerina-version.help]:
```suggestion
       supported language specification version.
```

The tool version seems to have been removed.

File: cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help (hunk 25 above)
Comment (MaryamZi) [reply cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help]: The following would have to be done
```suggestion
       ballerina test -a | --all [<options>] [--] [(--key=value)...]
```

**Comment by anupama-pathirage** [comment]
Merging this and will send the comments in a different PR

**Comment by MaryamZi** [comment]
> Merging this and will send the comments in a different PR

Suggestions will be addressed with https://github.com/ballerina-platform/ballerina-lang/pull/18924.

**Pull request merged**